code/modules/flufftext/Hallucination.dm

```diff
@@ -346,82 +397,122 @@ GLOBAL_LIST_INIT(hallucination_list, list(
-					addtimer(CALLBACK(target, /mob/.proc/playsound_local, source, 'sound/weapons/effects/searwall.ogg', 25, 1), rand(5,10))
-				sleep(rand(CLICK_CD_RANGE, CLICK_CD_RANGE + 6))
-				if(hits >= 4 && prob(70))
-					target.playsound_local(source, get_sfx("bodyfall"), 25, 1)
-					break
-		if("disabler")
-			var/hits = 0
-			for(var/i in 1 to rand(5, 10))
-				target.playsound_local(source, 'sound/weapons/taser2.ogg', 25, 1)
-				if(prob(50))
-					addtimer(CALLBACK(target, /mob/.proc/playsound_local, source, 'sound/weapons/tap.ogg', 25, 1), rand(5,10))
-					hits++
-				else
-					addtimer(CALLBACK(target, /mob/.proc/playsound_local, source, 'sound/weapons/effects/searwall.ogg', 25, 1), rand(5,10))
-				sleep(rand(CLICK_CD_RANGE, CLICK_CD_RANGE + 6))
-				if(hits >= 3 && prob(70))
-					target.playsound_local(source, get_sfx("bodyfall"), 25, 1)
-					break
+		if("disabler", "laser")
+			iterations_left = rand(5, 10)
 		if("esword")
+			iterations_left = rand(4, 8)
 			target.playsound_local(source, 'sound/weapons/saberon.ogg',15, 1)
-			for(var/i in 1 to rand(4, 8))
-				target.playsound_local(source, 'sound/weapons/blade1.ogg', 50, 1)
-				if(i == 4)
-					target.playsound_local(source, get_sfx("bodyfall"), 25, 1)
-				sleep(rand(CLICK_CD_MELEE, CLICK_CD_MELEE + 6))
-			target.playsound_local(source, 'sound/weapons/saberoff.ogg', 15, 1)
 		if("gun")
-			var/hits = 0
-			for(var/i in 1 to rand(3, 6))
-				target.playsound_local(source, 'sound/weapons/gun/shotgun/shot.ogg', 25, TRUE)
-				if(prob(60))
-					addtimer(CALLBACK(target, /mob/.proc/playsound_local, source, 'sound/weapons/pierce.ogg', 25, 1), rand(5,10))
-					hits++
-				else
-					addtimer(CALLBACK(target, /mob/.proc/playsound_local, source, "ricochet", 25, 1), rand(5,10))
-				sleep(rand(CLICK_CD_RANGE, CLICK_CD_RANGE + 6))
-				if(hits >= 2 && prob(80))
-					target.playsound_local(source, get_sfx("bodyfall"), 25, 1)
-					break
+			iterations_left = rand(3, 6)
 		if("stunprod") //Stunprod + cablecuff
+			process = FALSE
 			target.playsound_local(source, 'sound/weapons/egloves.ogg', 40, 1)
 			target.playsound_local(source, get_sfx("bodyfall"), 25, 1)
-			sleep(20)
-			target.playsound_local(source, 'sound/weapons/cablecuff.ogg', 15, 1)
+			addtimer(CALLBACK(target, /mob/.proc/playsound_local, source, 'sound/weapons/cablecuff.ogg', 15, 1), 20)
 		if("harmbaton") //zap n slap
+			iterations_left = rand(5, 12)
 			target.playsound_local(source, 'sound/weapons/egloves.ogg', 40, 1)
 			target.playsound_local(source, get_sfx("bodyfall"), 25, 1)
-			sleep(20)
-			for(var/i in 1 to rand(5, 12))
-				target.playsound_local(source, "swing_hit", 50, 1)
-				sleep(rand(CLICK_CD_MELEE, CLICK_CD_MELEE + 4))
+			next_action = 2 SECONDS
 		if("bomb") // Tick Tock
-			for(var/i in 1 to rand(3, 11))
+			iterations_left = rand(3, 11)
+
+	if (process)
+		START_PROCESSING(SSfastprocess, src)
+	else
+		qdel(src)
+
+/datum/hallucination/battle/process(delta_time)
+	next_action -= (delta_time * 10)
```

Comment:
So actions are in deciseconds? Does fastprocess support 11 deciseconds in contrast to 12?

Reply:
Fast process is fired every tick, I believe, and delta_time is given in seconds.

Reply:
It's on a wait of 2, so once every 2 ticks, delta_time is passed in like so 
```
if(QDELETED(thing))
			processing -= thing
		else if(thing.process(wait * 0.1) == PROCESS_KILL)
			// fully stop so that a future START_PROCESSING will work
			STOP_PROCESSING(src, thing)
		if (MC_TICK_CHECK)
			return
```
So in this case, it's 0.2

Reply:
Bit lost, what's the feedback here?

Reply:
ah ignore me, misread code